purchase_order_type_product_categ/views/purchase_order_views.xml

```diff
@@ -0,0 +1,29 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+    <data>
+        <record id="purchase_order_form" model="ir.ui.view">
+            <field name="name">purchase.order.form.inherit (in purchase_order_type_product_categ)</field>
+            <field name="model">purchase.order</field>
+            <field name="inherit_id" ref="purchase.purchase_order_form"/>
+            <field name="arch" type="xml">
+                <xpath expr="//field[@name='company_id']" position="after">
+                    <field name="product_category_id" invisible="1"/>
+                    <field name="product_category_child_ids" invisible="1"/>
+                </xpath>
+                <xpath expr="//tree/field[@name='product_id']" position="attributes">
+                    <attribute
+                            name="domain"
+                            operation="domain_add"
+                            condition="parent.product_category_id != False"
+                            join_operator="AND">
+                            [
+                                '|', 
+                                ('categ_id', '=', parent.product_category_id),
+                                ('categ_id', 'in', parent.product_category_child_ids), 
```

Comment:
Have you tried directly using `child_of` operator here?

Reply:
Mmm.. Its work, me bad